src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -619,4 +619,17 @@ public function getReturnStatementsAst()
 
         return $visitor->getReturnNodes();
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
+    {
+        $objectType = (new FindTypeFromAst())->__invoke(
+            $this->reflector->getContextFactory()->createForNamespace(
```

Comment:
Not sure - I think its pretty acceptable to catch exceptions in in this case (e.g. the [CMF chain router](https://github.com/symfony-cmf/routing/blob/master/src/ChainRouter.php#L176)).

Besides, if it can return NULL (f.e. instead of a `LocatedSource`), then I still don't see why the locator should instantiate the reflection class? Sorry, I realise that I am probably missing something here. Will have to investigate further.